### DISCUSSION TITLE
Solver blows up at high simulation times

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#34] [General discussion]
Hello all,

I am having a problem when running the pipeflow case for a long time (simulation time of 1 s). The case blows up suddenly in between and I am having no idea what is happening. I see maximum velocity values increase gradually until at a point it diverges.

I must run the case in smaller intervals so I can prevent the blow-up. Any ideas what is happening?

Also is there any way that I can remove older timesteps while running the case automatically? for instance I only need the last five saved timesteps.

P.S: I have attached the log file.
[out_6026484.txt](https://github.com/UvaCsl/HemoCell/files/7946406/out_6026484.txt)

**Comment by mojtabaAmiraslanpour** [comment]
So I have plotted values of max velocity with iteration and as you can see there are some very severe abnormalities in there.
![image](https://user-images.githubusercontent.com/15649053/151667591-54d4711f-fb1d-4bf4-8b75-7bc2a2eff0e7.png)

So I am not sure if there are any relaxation schemes or velocity capping in Hemocell that I can use to smooth these. Do you have any suggestions on how to make this work? It must be easy since you guys have run this problem before for up to 1s. I am stuck in about 0.4 s for the D32 case.

_I would appreciate it if you could help me with it._

**Comment by mojtabaAmiraslanpour** [comment]
I mention that there are multiple cell penetrations in the simulation domain after a few iterations:

![image](https://user-images.githubusercontent.com/15649053/151668467-6ad6ed53-1c01-42ca-9d76-06365982a4ba.png)
![image](https://user-images.githubusercontent.com/15649053/151668469-4e3e2ab1-9ea2-4791-bdc1-c2d20aeccbce.png)
![image](https://user-images.githubusercontent.com/15649053/151668470-4b82fd23-544f-48d3-bee5-cc01432b820f.png)

not sure what is causing this to happen!

Of course, I have used H 0.7 in packCells which upon starting the simulation decreases to about 0.3 due to RBC removal in boundaries. Also, I have used higher plt_ratio values to increase the number of PLTs artificially. Can these be effective on solver blow up?

**Comment by mojtabaAmiraslanpour** [comment]
Of course, I have used H 0.7 in packCells which upon starting the simulation decreases to about 0.3 due to RBC removal in boundaries. Also, I have used higher plt_ratio values to increase the number of PLTs artificially. Can these be effective on solver blow up?

**Comment by mojtabaAmiraslanpour** [comment]
Ok, I decreased plt_Ratio to 0.1 and it solved the problem.

**Comment by gzavo** [comment]
Good to see that you have solved it :)
In retrospect the overlapping cells cause numerical errors that accumulate over time. This is more severe for PLTs as they are set to be more rigid (=higher force response to deformation).
